extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,263 @@
+     * <li>{@link ChangeType#DELETE}: Deletes a running query by stopping the
+     * {@link QueryExecutor} service of the queryID in the event</li>
+     * <li>{@link ChangeType#UPDATE}: If the query is running and the update is
+     * to stop the query, stops the query. Otherwise, if the query is not
+     * running, it is removed.</li>
+     */
+    private class QueryManagerQueryChange implements QueryChangeLogListener {
+        private final String ryaInstanceName;
+
+        public QueryManagerQueryChange(final String ryaInstanceName) {
+            this.ryaInstanceName = requireNonNull(ryaInstanceName);
+        }
+
+        @Override
+        public void notify(final ChangeLogEntry<QueryChange> queryChangeEvent, final Optional<StreamsQuery> newQueryState) {
+            LOG.debug("New query change event.");
+            final QueryChange entry = queryChangeEvent.getEntry();
+
+            lock.lock();
+            try {
+
+                switch (entry.getChangeType()) {
+                    case CREATE:
+                        LOG.debug("Creating query event.");
+                        if (newQueryState.isPresent()) {
+                            runQuery(ryaInstanceName, newQueryState.get());
+                            LOG.info("Created new query: " + newQueryState.get().toString());
+                        }
+                        break;
+                    case DELETE:
+                        LOG.debug("delete query event.");
+                        if (newQueryState.isPresent()) {
+                            stopQuery(newQueryState.get().getQueryId());
+                            LOG.info("Deleted query: " + newQueryState.get().toString());
+                        } else {
+                            LOG.debug("Delete requested a query that does not exist yet.");
+                        }
+                        break;
+                    case UPDATE:
+                        LOG.debug("update query event.");
+                        if (!newQueryState.isPresent()) {
+                            LOG.debug("Query: " + entry.getQueryId() + " does not exist yet, cannot perform update.");
+                        } else {
+                            final StreamsQuery updateQuery = newQueryState.get();
+                            // if the query is currently inactive, and is updated to
+                            // be active, turn on.
+                            if (!updateQuery.isActive() && entry.getIsActive().or(false)) {
+                                final StreamsQuery newQuery = new StreamsQuery(entry.getQueryId(), entry.getSparql().get(),
+                                        entry.getIsActive().or(false));
+                                runQuery(ryaInstanceName, newQuery);
+                                LOG.info("Starting query: " + newQuery.toString());
+                            } else if (!updateQuery.isActive() && !entry.getIsActive().or(true)) {
```

Comment:
I don't think this logic is correct. You don't have enough information within this conditional to know if a change happened since you don't have the state prior to the change being applied. You know what the change was and the new state of the query after the change. You could ask the executor if it is currently running that query and stop it, though.

Reply:
so, if the new state is active, just try having the executor start/stop it?

Reply:
I don't honestly see anything wrong with that now that I think about it. The executor could be responsible for ignoring repetitive requests since it knows what is running anyway.

Reply:
Which kind of makes me think the edit we made for supplying the finished state of the query might not actually be needed.